lio_sam:
  # Topics
  # pointCloudTopic: "/velodyne_left_points"               # Point cloud data
  # imuTopic: "/imu/data"                         # IMU data
  pointCloudTopic: "/rslidar_points"               # Point cloud data
  imuTopic: "/gnss_imu"                         # IMU data
  odomTopic: "odometry/imu"                   # IMU pre-preintegration odometry, same frequency as IMU
  gpsTopic: "odometry/gps_manual"                   # GPS odometry topic from navsat, see module_navsat.launch file
  gpsRawTopic: "gps/fix"
  FOR_CHANGAN: false

  # Frames
  lidarFrame: "base_link"
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"
  keyframenumber: 10000
  containersize: 1000

  # GPS Settings
  useGps: false
  useImuHeadingInitialization: true           # if using GPS data, set to "true"
  useGpsElevation: true                      # if GPS elevation is bad, set to "false"
  gpsCovThreshold: 2.0                        # m^2, threshold for using GPS data
  poseCovThreshold: 0.01                      # m^2, threshold for using GPS data
  
  # Export settings
  savePCD: false                              # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/Downloads/KIT/"        # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

  # Save Path
  Odom_Path: "aaa/home/a/driver_ws/src/map_localization/results/"

  # Sensor Settings
  sensor: rslidar32                            # lidar sensor type, either 'velodyne' or 'ouster'
  N_SCAN: 32
  # sensor: velodyne
  # N_SCAN: 16                                  # number of lidar channel (i.e., 16, 32, 64, 128)
  Horizon_SCAN: 1800                          # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048)
  downsampleRate: 1                           # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1
  lidarMinRange: 1.0                          # default: 1.0, minimum lidar range to be used
  lidarMaxRange: 1000.0                       # default: 1000.0, maximum lidar range to be used
  USE_CORNER: false
 
  # lego
  groundScanInd: 7
  GroundAngleThreshold: 10
  segmentTheta: 1.0472
  segmentValidPointNum: 5
  segmentValidLineNum: 3
  segmentThresholdNum: 20

  # IMU Settings
  imuAccNoise: 3.9939570888238808e-03
  imuGyrNoise: 1.5636343949698187e-03
  imuAccBiasN: 6.4356659353532566e-05
  imuGyrBiasN: 3.5640318696367613e-05
  imuGravity: 9.805727
  imuRPYWeight: 0.01
  imuRate: 100.0
  imuQueSize: 100
  # imuTimeOffset: -0.08
  imuTimeOffset: 0.0
  # ieskf
  LIDAR_STD: 0.001

  # Ground Settings
  HeightThreshold: -1.6

  # Extrinsics (lidar -> IMU)
  # kaist
  # extrinsicTrans: [0.206692, 0.33898, -0.439121]
  # extrinsicRot: [-0.515105, 0.487008, -0.70533, -0.702383, -0.711468, 0.0217062, -0.491249, 0.506593, 0.708547]
  # extrinsicRPY: [-0.515105, 0.487008, -0.70533, -0.702383, -0.711468, 0.0217062, -0.491249, 0.506593, 0.708547]

  extrinsicTrans: [0, 0, 0]
  extrinsicRot: [0.999968, 0.007454, 0.002978,
                 -0.007457, 0.999972, 0.001121,
                  -0.002970, -0.001144, 0.999995]
  extrinsicRPY: [0.999968, 0.007454, 0.002978,
                 -0.007457, 0.999972, 0.001121,
                  -0.002970, -0.001144, 0.999995]

  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100

  # extract feature threshold
  MaxCornerSize: 20 #20
  MaxSurfSize: 60 #60
  MaxoptIteration: 15

  # voxel filter paprams
  odometrySurfLeafSize: 0.4                     # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerLeafSize: 0.2                    # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfLeafSize: 0.4                      # default: 0.4 - outdoor, 0.2 - indoor

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 0.0                            # meters
  rotation_tollerance: 1000                     # radians

  # CPU Params
  numberOfCores: 4                              # number of cores for mapping optimization
  mappingProcessInterval: 0.15                  # seconds, regulate mapping frequency

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 1.0   # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 0.2  # radians, regulate keyframe adding threshold
  surroundingKeyframeDensity: 2.0               # meters, downsample surrounding keyframe poses   
  surroundingKeyframeSearchRadius: 50.0         # meters, within n meters scan-to-map optimization (when loop closure disabled)

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
  surroundingKeyframeSize: 50                   # submap size (when loop closure enabled)
  historyKeyframeSearchRadius: 15.0             # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyframeSearchTimeDiff: 30.0           # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyframeSearchNum: 25                  # number of hostory key frames will be fused into a submap for loop closure
  historyKeyframeFitnessScore: 0.3              # icp threshold, the smaller the better alignment

  # Visualization
  globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 1.0           # meters, global map visualization cloud density
